sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryPartB.cs

```diff
@@ -123,6 +129,58 @@ private static void AddPropertiesToTelemetry(IDictionary<string, string> destina
             }
         }
 
+        private static void AddActivityLinksToPartCTags(IEnumerable<ActivityLink> links, ref AzMonList PartCTags)
+        {
+            string msLinks = "_MS.links";
+            int maxLength = 8192;
+            int prevIndex = 0;
+
+            if (links != null && links.Any())
+            {
+                var linksJson = new StringBuilder();
+                linksJson.Append('[');
+                foreach (var link in links)
+                {
+                    linksJson
+                        .Append('{')
+                        .Append("\"operation_Id\":")
+                        .Append('\"')
+                        .Append(link.Context.TraceId.ToHexString())
+                        .Append('\"')
+                        .Append(',');
+                    linksJson
+                        .Append("\"id\":")
+                        .Append('\"')
+                        .Append(link.Context.SpanId.ToHexString())
+                        .Append('\"');
+                    linksJson.Append("},");
+
+                    if (linksJson.Length > maxLength)
+                    {
+                        break;
+                    }
+
+                    prevIndex = linksJson.Length;
```

Comment:
Getting a length of stringbulider every time is costlier. We could decrement the constant link length from 8192 and looks out for its length is greater than 0.